framework/source/class/qx/core/MEvent.js

```diff
@@ -186,6 +186,31 @@ qx.Mixin.define("qx.core.MEvent",
     },
 
 
+    /**
+     * Creates and dispatches an event on this object; equivalent to fireEvent, except that it
+     * always returns a promise
+     *
+     * @param type {String} Event type to fire
+     * @param clazz {Class?qx.event.type.Event} The event class
+     * @param args {Array?null} Arguments, which will be passed to
+     *       the event's init method.
+     * @return {qx.Promise} a promise aggregated from the event handlers;
+     *  if the default was prevented, the promise is rejected
+     */
+    fireEventAsync : function(type, clazz, args)
+    {
+      if (!qx.core.Environment.get("qx.promise")) {
+        throw new Error(this.classname + ".fireEventAsync not supported because qx.promise==false");
+      }
+      
+      if (!this.$$disposed) {
+        return this.__Registration.fireEventAsync(this, type, clazz, args);
+      }
+
+      return qx.Promise.resolve(true);
```

Comment:
Why does this resolve if `this` is disposed? I'd think it should reject. It's a programmer error to fire an event on a disposed object...

Reply:
I agree, but that's how the ordinary, non-async `fireEvent` works so this is mirroring functionality but just wrapping it in a promise.

Reply:
Sigh.